### ISSUE TITLE
Naming of modules and functions

### ISSUE THREAD

**Issue by Rufflewind** [sympy/sympy#12782]
The current naming system where modules and functions have the same name makes it tricky to import the modules because functions always take precedence.  This has led to at least two StackOverflow questions:

  - https://stackoverflow.com/q/41214303
  - https://stackoverflow.com/a/22375385

There's a workaround (below), but it still doesn't let you import the module itself, just its members.  So you can't just blanket import a module into some local namespace.

```python
import sympy.printing.pretty as p
p.PrettyPrinter # doesn't work

import sympy.printing.pretty.pretty as p
p.PrettyPrinter # still doesn't work

from sympy.printing.pretty import pretty as p
p.PrettyPrinter # nice try

from sympy.printing.pretty.pretty import PrettyPrinter
PrettyPrinter # finally! -.-
```

I think it would be best to rename the modules at some point to avoid this messy situation.

**Comment by asmeurer** [comment]
Yeah, I've seen this issue before. The problem (I believe) is that `from module import submodule` only works because Python defines `submodule` in `module/__init__.py` automatically. But if you redefine `submodule` in there, say by importing a function by the same name, you lose the ability to access `submodule` from the `module` namespace. It's honestly pretty dumb that `import sympy.printing.pretty as p` assigns `p` to the function rather than the module (it's not like you can even do `import sympy.printing.StrPrinter`). It's one of those things that happens for technical reasons (`import as` is the same as `import` followed by an assignment). 

Stackoverflow shows the workarounds: either use `importlib` or `sys.modules`. 

Renaming would be a bit of an API break (especially for stuff that isn't included in `sympy/__init__.py`, but perhaps worth it). Alias modules is not a bad idea, but I wonder if it's really needed, given that there are workarounds.

**Comment by smichr** [comment]
Is it worthwhile to add a helper?

```python
def module(s):
    """return the module described by string s

    Examples
    ========

    >>> from sympy.utilities.misc import module
    >>> fu = module('sympy.simplify.fu')
    >>> 'TR1' in dir(fu)
    True
    """
    import importlib
    return importlib.import_module(s)
```
